src/core/CoreModuleDirector.ts

```diff
@@ -291,27 +330,30 @@ export class CoreModuleDirector {
   }
 
   public getSubscriptionOfTypeWithToken(
-    type: ModelName,
+    type: SubscriptionType,
     token: string,
   ): OSModel<SupportedSubscription> | undefined {
     logMethodCall('CoreModuleDirector.getSubscriptionOfTypeWithToken', {
       type,
       token,
     });
     switch (type) {
-      case ModelName.EmailSubscriptions: {
+      case SubscriptionType.Email: {
         const emailSubscriptions = this.getEmailSubscriptionModels();
         return Object.values(emailSubscriptions).find(
           (subscription) => subscription.data.token === token,
         );
       }
-      case ModelName.SmsSubscriptions: {
+      case SubscriptionType.SMS: {
         const smsSubscriptions = this.getSmsSubscriptionModels();
         return Object.values(smsSubscriptions).find(
           (subscription) => subscription.data.token === token,
         );
       }
-      case ModelName.PushSubscriptions: {
+      case SubscriptionType.ChromePush:
+      case SubscriptionType.SafariPush:
+      case SubscriptionType.SafariLegacyPush:
+      case SubscriptionType.FirefoxPush: {
```

Comment:
If we are making this change we should use the new `isPushSubscriptionType()` helper method you created.